modules/admin/src/main/java/org/motechproject/admin/internal/service/impl/SettingsServiceImpl.java

```diff
@@ -244,24 +246,25 @@ public void saveRawFile(MultipartFile file, String filename, long bundleId) {
         Enumeration<URL> enumeration = bundle.findEntries("", "*.properties", false);
         Map<String, Properties> allDefaultProperties = new LinkedHashMap<>();
 
-        while (enumeration.hasMoreElements()) {
-            InputStream is = null;
-            URL url = enumeration.nextElement();
-            try {
-                is = url.openStream();
-                Properties defaultBundleProperties = new Properties();
-                defaultBundleProperties.load(is);
-                if (!url.getFile().isEmpty()) {
-                    //We want to store plain filename, without unnecessary slash prefix
-                    allDefaultProperties.put(url.getFile().substring(1), defaultBundleProperties);
+        if (CollectionUtils.isNotEmpty(Collections.list(enumeration))) {
```

Comment:
@jhopen or @mkruszynski  Can you explain why adding this if solves all our problems? It seems that if this enumeration is empty but not null, it should just skip the while loop below and return an empty map. Can this be null? If yes, when? and is this a correct behaviour?